test/helpers/machineosconfigbuilder.go

```diff
@@ -21,81 +21,58 @@ func NewMachineOSConfigBuilder(name string) *MachineOSConfigBuilder {
+
+				Containerfile: []mcfgv1.MachineOSContainerfile{},
+				ImageBuilder: mcfgv1.MachineOSImageBuilder{
+					ImageBuilderType: mcfgv1.MachineOSImageBuilderType("PodImageBuilder"),
 				},
+				BaseImagePullSecret:     &mcfgv1.ImageSecretObjectReference{},
+				RenderedImagePushSecret: mcfgv1.ImageSecretObjectReference{},
 			},
 		},
 	}
 }
 
-func (m *MachineOSConfigBuilder) WithReleaseVersion(version string) *MachineOSConfigBuilder {
-	m.mosc.Spec.BuildInputs.ReleaseVersion = version
-	return m
-}
-
-func (m *MachineOSConfigBuilder) WithBaseOSImagePullspec(pullspec string) *MachineOSConfigBuilder {
-	m.mosc.Spec.BuildInputs.BaseOSImagePullspec = pullspec
-	return m
-}
-
 func (m *MachineOSConfigBuilder) WithMachineConfigPool(name string) *MachineOSConfigBuilder {
 	m.mosc.Spec.MachineConfigPool.Name = name
 	return m
 }
 
 func (m *MachineOSConfigBuilder) WithBaseImagePullSecret(name string) *MachineOSConfigBuilder {
-	m.mosc.Spec.BuildInputs.BaseImagePullSecret.Name = name
+	m.mosc.Spec.BaseImagePullSecret.Name = name
 	return m
 }
 
 func (m *MachineOSConfigBuilder) WithFinalImagePushSecret(name string) *MachineOSConfigBuilder {
 	return m.WithRenderedImagePushspec(name)
 }
 
-func (m *MachineOSConfigBuilder) WithContainerfile(arch mcfgv1alpha1.ContainerfileArch, content string) *MachineOSConfigBuilder {
-	m.mosc.Spec.BuildInputs.Containerfile = append(m.mosc.Spec.BuildInputs.Containerfile, mcfgv1alpha1.MachineOSContainerfile{
+func (m *MachineOSConfigBuilder) WithContainerfile(arch mcfgv1.ContainerfileArch, content string) *MachineOSConfigBuilder {
+	m.mosc.Spec.Containerfile = append(m.mosc.Spec.Containerfile, mcfgv1.MachineOSContainerfile{
 		ContainerfileArch: arch,
 		Content:           content,
 	})
 	return m
 }
 
 func (m *MachineOSConfigBuilder) WithRenderedImagePushspec(pushspec string) *MachineOSConfigBuilder {
-	m.mosc.Spec.BuildInputs.RenderedImagePushspec = pushspec
+	m.mosc.Spec.RenderedImagePushSpec = mcfgv1.ImageTagFormat(pushspec)
 	return m
 }
 
 func (m *MachineOSConfigBuilder) WithRenderedImagePushSecret(name string) *MachineOSConfigBuilder {
-	m.mosc.Spec.BuildInputs.RenderedImagePushSecret.Name = name
-	return m
-}
-
-func (m *MachineOSConfigBuilder) WithExtensionsImagePullspec(pullspec string) *MachineOSConfigBuilder {
-	m.mosc.Spec.BuildInputs.BaseOSExtensionsImagePullspec = pullspec
+	m.mosc.Spec.RenderedImagePushSecret.Name = name
 	return m
 }
 
 func (m *MachineOSConfigBuilder) WithCurrentImagePullspec(pullspec string) *MachineOSConfigBuilder {
-	m.mosc.Status.CurrentImagePullspec = pullspec
-	return m
-}
-
-func (m *MachineOSConfigBuilder) WithCurrentImagePullSecret(name string) *MachineOSConfigBuilder {
-	m.mosc.Spec.BuildOutputs.CurrentImagePullSecret.Name = name
+	m.mosc.Status.CurrentImagePullSpec = mcfgv1.ImageDigestFormat(pullspec)
 	return m
 }
 
+// QOCL:Is CurrentImagePullSecret gone? It wasn't listed in the story as one of the fields that was removed.
```

Comment:
in ref to https://github.com/openshift/machine-config-operator/pull/4756/commits/6064540dc3a033132a27ae975bc44efb67588e35#r1884535019

Reply:
I'm pretty sure it is. This value will now come from the global image pull secret instead.

Reply:
ack, updated https://issues.redhat.com/browse/MCO-1476 to reflect this